src/test/java/uk/gov/justice/hmpps/prison/service/OffenderTransactionHistoryServiceTest.java

```diff
@@ -0,0 +1,316 @@
+package uk.gov.justice.hmpps.prison.service;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.justice.hmpps.prison.api.model.OffenderTransactionHistoryDto;
+import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderTransactionHistory;
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.OffenderTransactionHistoryRepository;
+import uk.gov.justice.hmpps.prison.repository.jpa.model.Offender;
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.OffenderRepository;
+import uk.gov.justice.hmpps.prison.service.transformers.OffenderTransactionHistoryTransformer;
+
+import javax.lang.model.UnknownEntityException;
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.util.Optional;
+import java.util.List;
+import java.util.Collections;
+import java.util.Arrays;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@ExtendWith(MockitoExtension.class)
+public class OffenderTransactionHistoryServiceTest {
+
+    public static final String OFFENDER_NO = "A1111AA";
+
+    @Mock
+    private OffenderTransactionHistoryRepository repository;
+
+    @Mock
+    private OffenderRepository offenderRepository;
+
+    private OffenderTransactionHistoryService service;
+
+    @BeforeEach
+    public void setUp() {
+        service = new OffenderTransactionHistoryService("GBP", repository, offenderRepository);
+    }
+
+    @Test
+    public void When_getTransactionHistory_And_AccountCodeIsGiven_Then_CallRepositoryWithAccountCode() {
+
+        final Optional<String> accountCode  = Optional.of("spends");
```

Comment:
One for a later PR but the convention is to use final var unless it's not obvious what the type is.